### ISSUE TITLE
test_sslverify fails with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5747]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5747|
|Type|defect|
|Created|2012-07-08 20:56:18Z|

One test depends on hash ordering and fails with python -R: 

```
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/test/test_sslverify.py", line 254, in test_inspectCertificate
    "Public Key with Hash: ff33994c80812aa95a79cdb85362d054"])
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = ['Certificate For Subject:',
 '  Organizational Unit Name: Security',
 '             Locality Name: Boston',
 '              Country Name: US',
 '             Email Address: nobody@twistedmatrix.com',
 '         Organization Name: Twisted Matrix Labs',
 '               Common Name: example.twistedmatrix.com',
 '    State Or Province Name: Massachusetts',
 '',
 'Issuer:',
 '  Organizational Unit Name: Security',
 '             Locality Name: Boston',
 '              Country Name: US',
 '             Email Address: nobody@twistedmatrix.com',
 '         Organization Name: Twisted Matrix Labs',
 '               Common Name: example.twistedmatrix.com',
 '    State Or Province Name: Massachusetts',
 '',
 'Serial Number: 12345',
 'Digest: C4:96:11:00:30:C3:EC:EE:A3:55:AA:ED:8C:84:85:18',
 'Public Key with Hash: ff33994c80812aa95a79cdb85362d054']
b = ['Certificate For Subject:',
 '  Organizational Unit Name: Security',
 '         Organization Name: Twisted Matrix Labs',
 '               Common Name: example.twistedmatrix.com',
 '    State Or Province Name: Massachusetts',
 '              Country Name: US',
 '             Email Address: nobody@twistedmatrix.com',
 '             Locality Name: Boston',
 '',
 'Issuer:',
 '  Organizational Unit Name: Security',
 '         Organization Name: Twisted Matrix Labs',
 '               Common Name: example.twistedmatrix.com',
 '    State Or Province Name: Massachusetts',
 '              Country Name: US',
 '             Email Address: nobody@twistedmatrix.com',
 '             Locality Name: Boston',
 '',
 'Serial Number: 12345',
 'Digest: C4:96:11:00:30:C3:EC:EE:A3:55:AA:ED:8C:84:85:18',
 'Public Key with Hash: ff33994c80812aa95a79cdb85362d054']


twisted.test.test_sslverify.OpenSSLOptions.test_inspectCertificate
```

Attachments:

* [randomhash_sslverify.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7a2/7a2b6ce23d01e0f235e87d58e4e9da422f67c65e/fcadd0c708c8fdc580cb7b2f00f4aa1bbed59c7e.patch) (2231 bytes) - added by antoine on 2012-07-08 21:03:33Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5747 5747
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1341780978000000 1341780978000000
changetime__1341785300000000 1341785300000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_1"></a>|@pitrou commented|
|-|-|

Patch attached.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  I almost said something about the cost of re-sorting this same list of strings every time, but then I decided that would be a stupid review comment.  This patch looks fine, I'll apply it.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [34701]) Adjust DistinguishedName.inspect to have stable results regardless of dict item ordering.

Author: antoine
Reviewer: exarkun
Fixes: [#5747](https://github.com/twisted/twisted/issues/5747)
